test/e2e/expirable-airdrop.spec.ts

```diff
@@ -0,0 +1,188 @@
+  let toClaim = ethers.utils.parseUnits(String(amount));
+
+  const leaves = accounts.map((x) => {
+    let address = x.address;
+    return getLeaf(address, toClaim);
+  });
+
+  const tree = new MerkleTree(leaves, ethers.utils.keccak256, { sortPairs: true });
+  const root = tree.getHexRoot();
+
+  return [root, tree];
+}
+
+describe('Expirable airdrop', () => {
+  let nowTimestamp: number;
+  const oneMonth: number = 30 * 24 * 60 * 60; // one month in seconds
+  let expirationTimestamp: number;
+  let accounts: any[];
+  let deployer: any;
+  let root: any;
+  let tree: any;
+  let token: any;
+  let expirableAirdrop: any;
+  let airdropAmount: number = 10;
+  let snapshot: any;
+  const randomAddress: string = '0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045';
+  const randomAddress2: string = '0x2089035369B33403DdcaBa6258c34e0B3FfbbBd9';
+
+  before(async () => {
+    nowTimestamp = await time.latest();
+    expirationTimestamp = nowTimestamp + 3 * oneMonth; // 3 months
+    [deployer, ...accounts] = await ethers.getSigners();
+
+    [root, tree] = createMerkleRoot(accounts, airdropAmount);
+
+    // deploy erc20 mock
+    const Token = await ethers.getContractFactory('ERC20Mock');
+    token = await Token.deploy('Token', 'TKN');
+
+    // deploy contract
+    const ExpirableAirdrop = await ethers.getContractFactory('ExpirableAirdrop');
+    expirableAirdrop = await ExpirableAirdrop.deploy(deployer.address, token.address, expirationTimestamp, root);
+  });
+
+  when('deposit', () => {
+    let toDeposit: BigNumber;
+
+    given(async () => {
+      toDeposit = toUnit(airdropAmount * (accounts.length + 1));
+
+      // transfer some tokens
+      await token.mint(deployer.address, toDeposit);
+    });
+
+    then('airdrop is deposited', async () => {
+      // approve
+      await token.approve(expirableAirdrop.address, ethers.constants.MaxUint256);
+
+      // deposit
+      await expirableAirdrop.depositTokens(toDeposit);
+
+      // save snapshot for later
+      snapshot = await takeSnapshot();
+
+      // expect
+      expect(await token.balanceOf(expirableAirdrop.address)).equal(toDeposit);
+    });
+  });
+
+  when('claim', () => {
+    let alice: any;
+    let leaf: any;
+    let proof: any;
+
+    given(async () => {
+      alice = accounts[1];
+      leaf = getLeaf(alice.address, toUnit(airdropAmount));
+      proof = tree.getHexProof(leaf);
+    });
+
+    then('airdrop claimed', async () => {
+      // user claims airdrop
+      await expirableAirdrop.connect(alice).claim(alice.address, toUnit(airdropAmount), proof);
+
+      // expect
+      expect(await token.balanceOf(alice.address)).equal(toUnit(airdropAmount));
+    });
+
+    then('revert if already claimed', async () => {
+      await expect(expirableAirdrop.connect(alice).claim(alice.address, toUnit(airdropAmount), proof)).to.be.revertedWithCustomError(
+        expirableAirdrop,
+        'AlreadyClaimed'
+      );
+    });
+
```

Comment:
1. Could we add a test where another address claims in behalf if Alice, and the tokens are sent to her?
2. Could we add a test that reverts when the claim has expired?